x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -447,21 +443,35 @@ func verifyClosestProof(
 	return nil
 }
 
-// validateMiningDifficulty ensures that the relay's mining difficulty meets the
+// validateRelayDifficulty ensures that the relay's mining difficulty meets the
 // required minimum threshold.
 // TODO_TECHDEBT: Factor out the relay mining difficulty validation into a shared
 // function that can be used by both the proof and the miner packages.
-func validateMiningDifficulty(relayBz []byte, minRelayDifficultyBits uint64) error {
-	relayHash := servicetypes.GetHashFromBytes(relayBz)
-	relayDifficultyBits := protocol.CountHashDifficultyBits(relayHash)
+func validateRelayDifficulty(relayBz []byte, targetHash []byte) error {
+	relayHash := protocol.GetHashFromBytes(relayBz)
+
+	if len(targetHash) != sha256.Size {
```

Comment:
Ditto regarding `sha256.size` not being a semantic constant.

Reply:
I ended up doing something similar to the comment referenced above, but decided to make a separate var and const for the relay hasher and size. Altogether, pkt/crypto/protocol/hasher.go will look like this:

```go
package protocol

import "crypto/sha256"

const (
	RelayHasherSize = sha256.Size
	TrieHasherSize  = sha256.Size
	TrieRootSize    = TrieHasherSize + trieRootMetadataSize
	// TODO_CONSIDERATION: Export this from the SMT package.
	trieRootMetadataSize = 16
)

var (
	NewRelayHasher = sha256.New
	NewTrieHasher  = sha256.New
)
```